### PR TITLE
Support customized nonce for get measurement request.

### PR DESCRIPTION
**Pull request by longlongyang**
Currently, the Nonce in get measurement is generated by `get_random`. This patch add a new paramenter `spdm_nonce_struct` to API `send_receive_spdm_measurement` to support customized Nonce.

### CODE REVIEW

**Review by xiaoyuxlu — COMMENTED** [review]
why not add a new function?

**Review by xiaoyuxlu — APPROVED** [review]
LGTM

File: spdmlib/src/requester/get_measurements_req.rs [hunk 1]
```diff
@@ -283,12 +293,42 @@ impl RequesterContext {
...
+            out_total_number,
+            spdm_measurement_record_structure,
+            transcript_meas,
+        )
+        .await
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    #[maybe_async::maybe_async]
+    pub async fn send_receive_spdm_measurement_with_nonce(
```
Comment (jyao1) [inline spdmlib/src/requester/get_measurements_req.rs:313]: please just change "send_receive_spdm_measurement()"

Comment (longlongyang) [reply spdmlib/src/requester/get_measurements_req.rs]: > please just change "send_receive_spdm_measurement()"

Updated.

**Pull request merged**